### PR TITLE
NSQ Output plugin

### PR DESCRIPTION
**Pull request by jrxFive**
NSQ output plugin #309, following the NSQ methodology output is a producer
to one instance of NSQD. The go library does not accept array values. String
default for a Producer. Additionally service discovery is generally
done as a consumer.

Follows same methodology as Kafka Output without the tag reference.

### CODE REVIEW

**Comment by sparrc** [comment]
@jrxFive You've written this based on an outdated version of telegraf. The `Write` function doesn't take BatchPoints anymore, it takes `[]*client.Point`

You should rebase your changes from HEAD, you'll see that the kafka output is quite a bit different too. It should actually simplify the `Write` function, because you will not need to deal with a .Raw or any Tag inheritance and parsing.

**Comment by sparrc** [comment]
@jrxFive Thank you very much for this! It looks good other than needing to be rebased

**Comment by jrxFive** [comment]
Sorry should have done that prior. It's a lot simpler than before :). I'll commit the change soon.

File: scripts/docker-compose.yml [hunk 1]
```diff
@@ -47,3 +47,9 @@ aerospike:
     image: aerospike/aerospike-server
     ports:
         - "3000:3000"
+
+nsq:
+    image: nsqio/nsq
+    ports:
+        - "4150:4150"
+        - "4151:4151"
```
Comment (sparrc) [inline scripts/docker-compose.yml]: What does port 4151 do?

Comment (jrxFive) [inline scripts/docker-compose.yml]: 4150 is an TCP listener port, 4151 is for HTTP even though the plugin isnt using that feature should I remove it?

Comment (sparrc) [inline scripts/docker-compose.yml]: that's fine, you can keep it

**Comment by sparrc** [comment]
@jrxFive did you run the NSQ docker test? I'm getting a `connection refused` failure on OSX:

```
2015/10/27 13:12:38 INF    1 (192.168.99.100:4150) connecting to nsqd
2015/10/27 13:12:38 ERR    1 (192.168.99.100:4150) error connecting to nsqd - dial tcp 192.168.99.100:4150: getsockopt: connection refused
--- FAIL: TestConnectAndWrite (0.00s)
        Error Trace:    nsq_test.go:27
	Error:		No error is expected but got FAILED to send NSQD message: dial tcp 192.168.99.100:4150: getsockopt: connection refused

FAIL
FAIL	github.com/influxdb/telegraf/outputs/nsq	0.018s
```

**Comment by jrxFive** [comment]
Hmm I ran tests on a Linux VM and directly against a NSQ daemon. Weird I'll try on OSX to reproduce.

**Comment by sparrc** [comment]
@jrxFive I'm trying to debug, for some reason the NSQ container is not starting for me

**Comment by sparrc** [comment]
It looks like we're missing the "command", it needs to look like this in docker-compose

```
nsq:
    image: nsqio/nsq
    command: /nsqd
    ports:
        - "4150:4150"
        - "4151:4151"
```

**Comment by sparrc** [comment]
@jrxFive I'll make the change and merge this

**Comment by jrxFive** [comment]
Ahh good catch! Hmm why didnt this fail on the Circle-CI test? Its using Ubuntu Circle-CI? Wonder why OSX caught that though?

**Comment by sparrc** [comment]
CircleCI isn't currently running the "long" docker tests, see #293 

I need to find some time to get that env working on Circle

**Comment by sparrc** [comment]
Boy, that became a doozy. I decided to eliminate `docker-compose` from the unit tests, instead just running the docker containers directly from the Makefile. docker-compose seems to do a lot of caching that I wasn't able to disable.

I never really liked the docker-compose requirement anyways :-)

**Pull request closed without merging**